community/caddy/caddy.initd

```diff
@@ -5,9 +5,37 @@ command="/usr/sbin/caddy"
 command_args="$CADDY_OPTS"
 pidfile="/var/run/caddy.pid"
 command_background=yes
-start_stop_daemon_args="--user caddy --group caddy"
+command_user=caddy:caddy
+extra_commands="checkconfig"
+extra_started_commands="reload"
 
 depend() {
 	need net localmount
 	after firewall
 }
+
+checkconfig() {
+	ebegin "Checking configuration for $RC_SVCNAME"
+	start-stop-daemon --start --exec $command \
```

Comment:
This is a bit unusual.
Further, I'm planning on rewriting this thing to use supervise-daemon later, and would appreciate having an easy transition (not a problem in `reload()` since supervise-daemon is also signal-aware).

Since we're hard-coding the user/group, it may be appropriate to do this using su(1).
Alternatively, since we specify an explicit `-conf` by default, it could be appropriate to just run the validation sequence as root (it's a read-only operation, after all).